src/media/headset.ts

```diff
@@ -0,0 +1,71 @@
+import { Observable } from 'rxjs';
+import GenesysCloudWebrtcSdk from '../client';
+import HeadsetService, { ConsumedHeadsetEvents, VendorImplementation} from 'softphone-vendor-headsets';
+
+export class SdkHeadset {
+    private sdk: GenesysCloudWebrtcSdk;
+    private headsetLibrary: HeadsetService;
+    headsetEvents$: Observable<ConsumedHeadsetEvents>;
+
+    constructor(sdk: GenesysCloudWebrtcSdk) {
+        this.sdk = sdk;
+        this.headsetLibrary = HeadsetService.getInstance({ logger: console });
+        this.headsetEvents$ = this.headsetLibrary.headsetEvents$;
+    }
+
+    updateAudioInputDevice (newMicId: string): void {
+        const completeDeviceInfo = this.sdk.media.findCachedDeviceByIdAndKind(newMicId, 'audioinput');
+        this.headsetLibrary.activeMicChange(completeDeviceInfo.label.toLowerCase());
+    }
+
+    getCurrentSelectedImplementation (): VendorImplementation {
+        return this.headsetLibrary.selectedImplementation;
+    }
+
+    showRetry (): boolean {
+        const selectedImplementation = this.getCurrentSelectedImplementation();
+        if (selectedImplementation.disableRetry) {
```

Comment:
Could `selectedImplementation` ever be falsy here? Same question for all the functions below.

Reply:
I suppose it could. If an unsupported device is selected, I believe selectedImplementation becomes null. However, if I remember correctly, the div in which `showRetry` and `retryConnection` are important won't even show up if selectedImplementation is falsy